src/key.cpp

```diff
@@ -227,10 +227,23 @@ class CECKey {
     }
 
     bool Verify(const uint256 &hash, const std::vector<unsigned char>& vchSig) {
-        // -1 = error, 0 = bad sig, 1 = good
-        if (ECDSA_verify(0, (unsigned char*)&hash, sizeof(hash), &vchSig[0], vchSig.size(), pkey) != 1)
+        if (vchSig.empty())
             return false;
-        return true;
+
+        // New versions of OpenSSL will reject non-canonical DER signatures. de/re-serialize first.
+        unsigned char *norm_der = NULL;
+        ECDSA_SIG *norm_sig = ECDSA_SIG_new();
```

Comment:
I think this should be "...= NULL;" - it's nearly 1am, can't finish testing tonight, have flagged for Bitcoin devs to look at.